### ISSUE TITLE
RecordState Ignored

### ISSUE THREAD

**Issue by PaulWoitaschek** [NYTimes/Store#284]
I tried to implement the persister.

At first access, my database has no entries. So I return `RecordState.MISSING`.

I expect the store to make a request in that case. But instead it calls my persister (which returns an empty list as there are no entries yet) and delivers the empty list.

**Comment by digitalbuddha** [comment]
You're right it's not working as expected, seems like stale is being handled correctly but not missing. I'll take a look tonight. Thank you for reporting.

**Comment by ramonaharrison** [comment]
It looks like an appropriate way to handle this could be by returning `Maybe.empty()` from `RealInternalStore#disk` when the persister is a `RecordProvider` with `RecordState.MISSING` -- this would trigger the store to switch to `fetch()` before calling the persister.

We could add a helper method for this to the `StoreUtil` class, similar to the existing method for determining if the persister is stale.
